galgebra/ga.py

```diff
@@ -1263,52 +1272,67 @@ def wedge(self, A, B):
             return 0
         return update_and_substitute(A, B, self.wedge_product_basis_blades, self.wedge_table_dict)
 
-    def dot(self, A, B):
-        r"""
-        Inner product ``|``, ``<``, or ``>``
-
-        Let :math:`A = a + A'` and :math:`B = b + B'` where :math:`a` and
-        :math:`b` are the scalar parts of :math:`A` and :math:`B`, and
-        :math:`A'` and :math:`B'` are the remaining parts of :math:`A` and
-        :math:`B`. Then we have:
-
-        .. math::
-
-            (a+A') \rfloor (b+B') &= a(b+B') + A' \rfloor B' \\
-            (a+A') \lfloor (b+B') &= b(a+A') + A' \lfloor B'
 
-        We use these relations to reduce :math:`A \rfloor B` (``A<B``) and 
-        :math:`A \lfloor B` (``A>B``).
-        """
+    def _dot(self, A, B, mode):
         if A == 0 or B == 0:
             return 0
         if self.is_ortho:
-            dot_product_basis_blades = self.dot_product_basis_blades
+            dot_product_basis_blades = lambda x: self.dot_product_basis_blades(x, mode=mode)
         else:
-            dot_product_basis_blades = self.non_orthogonal_dot_product_basis_blades
+            dot_product_basis_blades = lambda x: self.non_orthogonal_dot_product_basis_blades(x, mode=mode)
 
-        if self.dot_mode == '|':  # Hestenes dot product
+        if mode == '|':  # Hestenes dot product
             A = self.remove_scalar_part(A)
             B = self.remove_scalar_part(B)
             return update_and_substitute(A, B, dot_product_basis_blades, self.dot_table_dict)
-        elif self.dot_mode == '<' or self.dot_mode == '>':
+        elif mode == '<' or mode == '>':
+            r"""
+            Let :math:`A = a + A'` and :math:`B = b + B'` where :math:`a` and
+            :math:`b` are the scalar parts of :math:`A` and :math:`B`, and
+            :math:`A'` and :math:`B'` are the remaining parts of :math:`A` and
+            :math:`B`. Then we have:
+
+            .. math::
+
+                (a+A') \rfloor (b+B') &= a(b+B') + A' \rfloor B' \\
+                (a+A') \lfloor (b+B') &= b(a+A') + A' \lfloor B'
+
+            We use these relations to reduce :math:`A \rfloor B` (``A<B``) and 
+            :math:`A \lfloor B` (``A>B``).
+            """
```

Comment:
This isn't a real docstring anyway it seems - it describes implementation details, which don't need to appear on readthedocs

Reply:
ok